src/oci_image/images.js

```diff
@@ -0,0 +1,316 @@
+import { PackageURL } from "packageurl-js";
+import { Platform } from "./platform.js";
+import { getImageDigests, getImagePlatform } from "./utils.js";
+
+/**
+* Helper class for parsing docker repository/image names:
+*
+* - If the first part before the slash contains a "." or a ":" it is considered to be a registry URL
+* - A last part starting with a ":" is considered to be a tag
+* - The rest is considered the repository name (which might be separated via slashes)
+*
+* Example of valid names:
+*
+* - consol/tomcat-8.0
+* - consol/tomcat-8.0:8.0.9
+* - docker.consol.de:5000/tomcat-8.0
+* - docker.consol.de:5000/jolokia/tomcat-8.0:8.0.9
+*/
+export class Image {
+	static nameComponentRegexp = '[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?';
```

Comment:
It's not a good practice to name variables/constants after the type they have. Also if they're supposed to be immutable you can use snake case for all. I suggest the following.

```suggestion
	static NAME_COMP_PATTERN = '[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?';
	static DOMAIN_COMP_PATTERN = '(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])';
	static NAME_COMP_REGEX = new RegExp(this.NAME_COMP_PATTERN);
	static IMAGE_NAME_REGEX = new RegExp(this.NAME_COMP_PATTERN + '(?:(?:/' + this.NAME_COMP_PATTERN + ')+)?');
	static DOMAIN_REGEX = new RegExp('^' + this.DOMAIN_COMP_PATTERN + '(?:\\.' + this.DOMAIN_COMP_PATTERN + ')*(?::[0-9]+)?$');
	static TAG_REGEX = new RegExp('^[\\w][\\w.-]{0,127}$');
	static DIGEST_REGEX = new RegExp('^sha256:[a-z0-9]{32,}$');
```